source/session.ts

```diff
@@ -823,14 +919,16 @@ export class Session {
     const fileSize = data.size || file.size;
     const componentId = data.id || uuidV4();
     const componentLocationId = uuidV4();
-    let url;
-    let headers;
+    let url: string;
+    let headers: Record<string, string> = {};
 
-    const updateOnProgressCallback = (oEvent) => {
+    const updateOnProgressCallback = (
+      oEvent: ProgressEvent<XMLHttpRequestEventTarget>
+    ) => {
       let progress = 0;
 
       if (oEvent.lengthComputable) {
-        progress = parseInt((oEvent.loaded / oEvent.total) * 100, 10);
+        progress = Math.round((oEvent.loaded / oEvent.total) * 100);
```

Comment:
`Math.floor` may be more appropriate for progress number.
I think it's better to return 99% for 99.6 instead of 100%.

Reply:
thanks!